lib/renderers/0-3.js

```diff
@@ -0,0 +1,340 @@
+      case LIST_SECTION_TYPE:
+        return this.renderListSection(section);
+      case CARD_SECTION_TYPE:
+        return this.renderCardSection(section);
+      default:
+        throw new Error(`Cannot render mobiledoc section of type "${type}"`);
+    }
+  }
+
+  renderMarkersOnElement(element, markers) {
+    let elements = [element];
+    let currentElement = element;
+
+    for (let i=0, l=markers.length; i<l; i++) {
+      let marker = markers[i];
+      let [type, openTypes, closeCount, value] = marker;
+
+      for (let j=0, m=openTypes.length; j<m; j++) {
+        let markerType = this.markerTypes[openTypes[j]];
+        let [tagName] = markerType;
+        if (isValidMarkerType(tagName)) {
+          let openedElement = createElementFromMarkerType(markerType);
+          appendChild(currentElement, openedElement);
+          elements.push(openedElement);
+          currentElement = openedElement;
+        } else {
+          closeCount--;
+        }
+      }
+
+      switch (type) {
+        case MARKUP_MARKER_TYPE:
+          appendChild(currentElement, createTextNode(value));
+          break;
+        case ATOM_MARKER_TYPE:
+          appendChild(currentElement, this._renderAtom(value));
+          break;
+        default:
+          throw new Error(`Unknown markup type (${type})`);
+      }
+
+      for (let j=0, m=closeCount; j<m; j++) {
+        elements.pop();
+        currentElement = elements[elements.length - 1];
+      }
+    }
+  }
+
+  renderListItem(markers) {
+    const element = createElement('li');
+    this.renderMarkersOnElement(element, markers);
+    return element;
+  }
+
+  renderListSection([type, tagName, listItems]) {
+    if (!isValidSectionTagName(tagName, LIST_SECTION_TYPE)) {
+      return;
+    }
+    const element = createElement(tagName);
+    listItems.forEach(li => {
+      appendChild(element, (this.renderListItem(li)));
+    });
+    return element;
+  }
+
+  renderImageSection([type, src]) {
+    let element = createElement(IMAGE_SECTION_TAG_NAME);
+    element.src = src;
+    return element;
+  }
+
+  findCard(name) {
+    for (let i=0; i < this.cards.length; i++) {
+      if (this.cards[i].name === name) {
+        return this.cards[i];
+      }
+    }
+    if (name === ImageCard.name) {
+      return ImageCard;
+    }
+    return this._createUnknownCard(name);
+  }
+
+  _findCardByIndex(index) {
+    let cardType = this.cardTypes[index];
+    if (!cardType) {
+      throw new Error(`No card definition found at index ${index}`);
+    }
+
+    let [ name, payload ] = cardType;
+    let card = this.findCard(name);
+
+    return {
+      card,
+      payload
+    };
+  }
+
+  _createUnknownCard(name) {
+    return {
+      name,
+      type: RENDER_TYPE,
+      render: this.unknownCardHandler
+    };
+  }
+
+  _createCardArgument(card, payload={}) {
+    let env = {
+      name: card.name,
+      isInEditor: false,
+      onTeardown: (callback) => this._registerTeardownCallback(callback)
```

Comment:
Somehow I really think these `createCard` and atom hooks should be shared between the modes. They are not mobiledoc-version specific, they are just cards or atoms.